backend/pkg/config/config.go

```diff
@@ -0,0 +1,202 @@
+	}
+
+	return nil
+}
+
+// Parse Loads the config from flags and env.
+// env vars should start with HEADLAMP_CONFIG_ and use _ as separator
+// If a value is set both in flags and env then env takes priority.
+// eg:
+// export HEADLAMP_CONFIG_PORT=2344
+// go run ./cmd --port=3456
+// the value of port will be 2344.
+func Parse(args []string) (*Config, error) {
+	var config Config
+
+	f := flagset()
+
+	k := koanf.New(".")
+
+	if args == nil {
+		args = []string{}
+	} else if len(args) > 0 {
+		args = args[1:]
+	}
+	// Parse args
+	if err := f.Parse(args); err != nil {
+		return nil, fmt.Errorf("error parsing flags: %w", err)
+	}
+
+	// Load config from args
+	if err := k.Load(basicflag.Provider(f, "."), nil); err != nil {
+		return nil, fmt.Errorf("error loading config from flags: %w", err)
+	}
+
+	// Load config from env
+	if err := k.Load(env.Provider("HEADLAMP_CONFIG_", ".", func(s string) string {
+		return strings.ReplaceAll(strings.ToLower(strings.TrimPrefix(s, "HEADLAMP_CONFIG_")), "_", "-")
+	}), nil); err != nil {
+		return nil, fmt.Errorf("error loading config from env: %w", err)
+	}
+
+	if err := k.Unmarshal("", &config); err != nil {
+		return nil, fmt.Errorf("error unmarshal config: %w", err)
+	}
+
+	// Validate parsed config
+	if err := config.Validate(); err != nil {
+		return nil, err
+	}
+
+	kubeConfigPath := ""
+
+	// If we don't have a specified kubeConfig path, and we are not running
+	// in-cluster, then use the default path.
+	if config.KubeConfigPath != "" {
+		kubeConfigPath = config.KubeConfigPath
+	} else if !config.InCluster {
+		kubeConfigEnv := os.Getenv("KUBECONFIG")
+		if kubeConfigEnv != "" {
+			kubeConfigPath = kubeConfigEnv
+		} else {
+			kubeConfigPath = GetDefaultKubeConfigPath()
+		}
+	}
+
+	config.KubeConfigPath = kubeConfigPath
+
+	return &config, nil
+}
+
+func flagset() *flag.FlagSet {
+	f := flag.NewFlagSet("config", flag.ContinueOnError)
+
+	f.Bool("in-cluster", false, "Set when running from a k8s cluster")
+	f.Bool("dev", false, "Allow connections from other origins")
+	f.Bool("insecure-ssl", false, "Accept/Ignore all server SSL certificates")
+
+	f.String("kubeconfig", "", "Absolute path to the kubeconfig file")
+	f.String("html-static-dir", "", "Static HTML directory to serve")
+	f.String("plugins-dir", defaultPluginDir(), "Specify the plugins directory to build the backend with")
+	f.String("base-url", "", "Base URL path. eg. /headlamp")
+	f.Uint("port", defaultPort, "Port to listen from")
+	f.String("proxy-urls", "", "Allow proxy requests to specified URLs")
+
+	f.String("oidc-client-id", "", "ClientID for OIDC")
+	f.String("oidc-client-secret", "", "ClientSecret for OIDC")
+	f.String("oidc-idp-issuer-url", "", "Identity provider issuer URL for OIDC")
+	f.String("oidc-scopes", "profile,email",
+		"A comma separated list of scopes needed from the OIDC provider")
+
+	return f
+}
+
+// Gets the default plugins-dir depending on platform.
+func defaultPluginDir() string {
```

Comment:
Should this really be defined here? I'd assume this belongs to plugins.go . Maybe it does belong here, but I am wondering how much of this is configuration and how much is a default value (same for the default kube config).

Reply:
These are default configuration values, so it makes sense to reside in the `config` package.